### ISSUE TITLE
generateNextManifest in "afterEmit" hook

### ISSUE THREAD

**Issue by derekdowling** [hanford/next-offline#112]
**TL:DR;**
Is it possible to adjust the `InlineNextPrecacheManifestPlugin` plugin to execute `generateNextManifest` during an earlier Webpack phase such as `afterEmit` rather than as part of the `done` phase as seen here: https://github.com/hanford/next-offline/blob/master/plugin.js#L14-L24? Waiting until that phase makes the output of `withOffline` incompatible with other build cleanup Webpack plugins.

**Full Problem Description**
1. I am required to utilize `importWorkboxFrom` with a `local` value since I can only utilize a 1st party CDN for serving assets and need to manage this myself.
2. The 1st party CDN serves Next assets from `/<hash>_next/static` rather than `/static`.
3. The only way to make this work with Workbox is to utilize the `importsDirectory` option with a value of `/<hash>_next/static/` which causes the files to be output to `.next/<hash>_next/static/` which is outside the normal `.next/static` directory. There is no way around this because `importsDirectory` is also used as the root path to the Workbox library in the SW generated by this plugin.
4. To make the post-build process manageable I want to move the SW files back to the normal `.next/static` folder so that it can utilize the existing CDN asset placement logic. This can't trivially be solved with a post-build shell command because I need to know the build hash to move the correct folder in `.next`. I am also having difficulty using a custom Webpack file moving plugin since `InlineNextPrecacheManifestPlugin` causes the SW's PrecacheManifest to be built in the last possible phase of Webpack's build: `done`. As a result there is a race between my custom file moving plugin and `InlineNextPrecacheManifestPlugin` that cannot be trivially solved as far as I can tell.

Perhaps I am missing something trivial? Looking forward to any advice in this regard.

**Comment by hanford** [comment]
Hm. This isn't really a use case I've had to support or work through on my own, so I'm not 100% certain what the fix would be.

we could definitely add an option to make next-offline execute at different phases of webpack though.

If you'd like to contribute maybe add a flag a line in the PR and open a PR. Otherwise I can do it, but will hold off until you get back to me
